### PR TITLE
VIH-8547 New Reconcile Scheduler job

### PR DESCRIPTION
**Pull request by khizarkhanu-hmcts**
Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/VIH-8547

### Change description ###
New Scheduler job for Reconciliation of Audio files with Wowza storage with conferences on the day.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: SchedulerJobs/SchedulerJobs.Services/ReconcileHearingAudioService.cs [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using VideoApi.Client;
+
+namespace SchedulerJobs.Services
+{
+
+    public interface IReconcileHearingAudioService
```
Comment (PrasannaDommalapati) [inline SchedulerJobs/SchedulerJobs.Services/ReconcileHearingAudioService.cs:12]: Could you please extract interface to separate file?

File: SchedulerJobs/SchedulerJobs.Services/ReconcileHearingAudioService.cs [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+                        audioMissingList.Add(item.Name.ToString());
+                        _logger.LogInformation("ReconcileAudiorecordingsWithConferencesAsync - missing wowza audio or empty files for conferences - " + item.Name + "  Exception from Reconciliation - " + ex.Message);
+
+                    }
+                }
+                _logger.LogInformation("ReconcileAudiorecordingsWithConferencesAsync - Completed");
+            }
+            catch (Exception ex)
+            {
+                _logger.LogInformation("ReconcileAudiorecordingsWithConferencesAsync - Exception" + ex.StackTrace);
```
Comment (PrasannaDommalapati) [inline SchedulerJobs/SchedulerJobs.Services/ReconcileHearingAudioService.cs:71]: Change to log error please

File: SchedulerJobs/SchedulerJobs.Services/ReconcileHearingAudioService.cs [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+                    {
+                        var result = await _videoApiClient.ReconcileAudioFilesInStorageAsync(filename_prefix, item.Count);
+
+                        _logger.LogInformation("ReconcileAudiorecordingsWithConferencesAsync - Processing conferences - result " + result.ToString());
+
+                    }
+                    catch (Exception ex)
+                    {
+                        audioMissingList.Add(item.Name.ToString());
+                        _logger.LogInformation("ReconcileAudiorecordingsWithConferencesAsync - missing wowza audio or empty files for conferences - " + item.Name + "  Exception from Reconciliation - " + ex.Message);
```
Comment (PrasannaDommalapati) [inline SchedulerJobs/SchedulerJobs.Services/ReconcileHearingAudioService.cs:63]: Change to log error

File: SchedulerJobs/SchedulerJobs.UnitTests/Services/ReconcileHearingAudioServiceTests.cs [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+using SchedulerJobs.Services;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using VideoApi.Client;
+using VideoApi.Contract.Responses;
+
+namespace SchedulerJobs.UnitTests.Services
+{
+    public class ReconcileHearingAudioServiceTests
```
Comment (PrasannaDommalapati) [inline SchedulerJobs/SchedulerJobs.UnitTests/Services/ReconcileHearingAudioServiceTests.cs:13]: Logger calls not verified

**Review by PrasannaDommalapati — CHANGES_REQUESTED** [review]

**Review by krishnapottekkat-hmcts — DISMISSED** [review]
LGTM

File: SchedulerJobs/SchedulerJobs.UnitTests/Services/ReconcileHearingAudioServiceTests.cs (hunk 4 above)
Comment (khizarkhanu-hmcts) [reply SchedulerJobs/SchedulerJobs.UnitTests/Services/ReconcileHearingAudioServiceTests.cs:14]: as discussed these are not required to be verified per functionality

File: SchedulerJobs/SchedulerJobs.Services/ReconcileHearingAudioService.cs (hunk 3 above)
Comment (khizarkhanu-hmcts) [reply SchedulerJobs/SchedulerJobs.Services/ReconcileHearingAudioService.cs]: fixed

File: SchedulerJobs/SchedulerJobs.Services/ReconcileHearingAudioService.cs (hunk 2 above)
Comment (khizarkhanu-hmcts) [reply SchedulerJobs/SchedulerJobs.Services/ReconcileHearingAudioService.cs]: fixed

File: SchedulerJobs/SchedulerJobs.Services/ReconcileHearingAudioService.cs (hunk 1 above)
Comment (khizarkhanu-hmcts) [reply SchedulerJobs/SchedulerJobs.Services/ReconcileHearingAudioService.cs]: fixed

**Review by PrasannaDommalapati — APPROVED** [review]
Well done!!

**Review by krishnapottekkat-hmcts — APPROVED** [review]

**Pull request merged**